packages/jest-mock-apollo/src/test/index.test.tsx

```diff
@@ -2,19 +2,26 @@ import * as React from 'react';
 import {graphql} from '@shopify/react-apollo';
 
 import {mount} from 'enzyme';
+import {readFileSync} from 'fs';
+import * as path from 'path';
 
-import {registerSchema, createGraphQLClient} from '..';
+import GraphQLClientFactory, {GraphQLMock, GraphQLClientOptions} from '..';
 import unionOrIntersectionTypes from './fixtures/schema-unions-and-interfaces.json';
 import petQuery from './fixtures/PetQuery.graphql';
 
 // setup
-registerSchema({
-  schemaBuildPath: './fixtures/schema.graphql',
+const graphQLClientFactory = new GraphQLClientFactory({
+  schemaSrc: readFileSync(
+    path.resolve(__dirname, './fixtures/schema.graphql'),
+    'utf8',
+  ),
   unionOrIntersectionTypes,
 });
 
+const createGraphQLClient = graphQLClientFactory.create;
```

Comment:
Should we have separate tests for `graphQLClientFactory` (or whatever it ends up called) then the classes that it outputs? That way we could have one test suite that makes sure the factory configures things correctly, and another that makes sure the mock link functions correctly.

Reply:
Ideally, yes. There's already an issue for improving these tests, but I believe that should be done separately from this PR: https://github.com/Shopify/quilt/issues/30